File: 7kyu/(7 kyu) Two to One/(7 kyu) Two to One.py
# #1
# def longest(s1, s2):
#     res = s1 + s2
#     res = list(res)
#     res = dict.fromkeys(res)
#     res = sorted(res)
#     res = ''.join(res)
#     return res
# #2
# def longest(s1, s2):
#     return ''.join(sorted(dict.fromkeys(s1 + s2)))

# #3


def longest(s1, s2):
    return ''.join(sorted((set(s1 + s2))))

File: 7kyu/(7 kyu) Two to One/(7 kyu) Two to One.rs
// #1
// fn longest(a1: &str, a2: &str) -> String {
//     let s = format!("{}{}", a1, a2);
//     let mut v = s.chars().collect::<Vec<char>>();
//     v.sort();
//     let mut res = String::new();
//     for c in v.iter() {
//         if !res.contains(&c.to_string()) {
//             res.push(*c);
//         }
//     }
//     res.to_string()
// }
// #2
fn longest(a1: &str, a2: &str) -> String {
    let mut s = format!("{}{}", a1, a2).chars().collect::<Vec<char>>();
    s.sort();
    s.dedup();
    s.into_iter().collect()
}
